llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -231,6 +231,48 @@ static KernelMapEntryScope selectDeviceCodeSplitScopeAutomatically(Module &M) {
   return Scope_PerModule;
 }
 
+// Collect all the dependencies for the function.
+static void collectFunctionCallGraphNodes(llvm::Function *Func,
+                                          SetVector<const GlobalValue *> &GVs) {
+  if (!GVs.count(Func))
+    GVs.insert(Func);
```

Comment:
What is the point of a separate check if an element is present in `set` if it is anyway done by `insert` to avoid duplicates?